components/script/dom/document.rs

```diff
@@ -1469,35 +1469,20 @@ impl Document {
             // however *when* we do it is up to us.
             // Here, we're dispatching it after the key event so the script has a chance to cancel it
             // https://www.w3.org/Bugs/Public/show_bug.cgi?id=27337
-            match keyboard_event.key {
-                Key::Character(ref letter)
-                    if letter == " " && keyboard_event.state == KeyState::Up =>
-                {
-                    let maybe_elem = target.downcast::<Element>();
-                    if let Some(el) = maybe_elem {
-                        synthetic_click_activation(
-                            el,
-                            false,
-                            false,
-                            false,
-                            false,
-                            ActivationSource::NotFromClick,
-                        )
-                    }
+            if (keyboard_event.key == Key::Enter && keyboard_event.state == KeyState::Up) ||
+                (keyboard_event.code == Code::Space && keyboard_event.state == KeyState::Down)
```

Comment:
In the code before your change, it was `KeyState::Up` for both spaces and enters, is it intended that you now check for `KeyState::Down` for spaces?

Reply:
Oh, no. It's not intended. Fixed.